batch_job/src/amazon_batch_script.sh

```diff
@@ -0,0 +1,196 @@
+#!/bin/sh
+# 
+# File:   amazon_batch_script.sh
+# Author: Kyle Sweeney
+#
+# Created on Jul 7, 2017, 1:48:40 PM
+#
+
+if [ -z "$1" ]; then
+	echo "Empty number of CPUs requested."
+	echo "Correct Ussage: ./makeflow_amazon_batch_setup [desired_num_cpus] [min_number_cpus] [max_number_cpus] (config-output)"
+	exit -17
+fi
+
+if [ -z "$2" ]; then
+	echo "Empty number of CPUs requested."
+	echo "Correct Ussage: ./makeflow_amazon_batch_setup [desired_num_cpus] [min_number_cpus] [max_number_cpus] (config-output)"
+	exit -17
+fi
+
+if [ -z "$3" ]; then
+	echo "Empty number of CPUs requested."
+	echo "Correct Ussage: ./makeflow_amazon_batch_setup [desired_num_cpus] [min_number_cpus] [max_number_cpus] (config-output)"
+	exit -17
+fi
+
+cpus=$1
+min_cpus=$2
+max_cpus=$3
+time=$(date +%s)
+
+if [ $cpus -lt $min_cpus ]; then
+	echo "Desired number of cpus must be greater than or equal to minimum number of cpus"
+	echo "Correct Ussage: ./makeflow_amazon_batch_setup [desired_num_cpus] [min_number_cpus] [max_number_cpus] (config-output)"
+        exit -17
+fi
+
+if [ $cpus -gt $max_cpus ]; then
+	echo "Desired number of cpus must be less than or equal to maximum number of cpus"
+        echo "Correct Ussage: ./makeflow_amazon_batch_setup [desired_num_cpus] [min_number_cpus] [max_number_cpus] (config-output)"
+        exit -17
+fi
+
+if [ $max_cpus -lt $min_cpus ]; then
+	echo "Minimum number of cpus must be less than or equal to maximum number of cpus"
+        echo "Correct Ussage: ./makeflow_amazon_batch_setup [desired_num_cpus] [min_number_cpus] [max_number_cpus] (config-output)"
+        exit -17
+fi
+
+if [ -n "$4" ]; then
+	outputfile="$4"
+else
+	outputfile="makeflow_amazon_batch.config"
+fi
+
+if [ -e $outputfile ]; then
+	echo "File: $outputfile already exists! Please cleanup/delete first"
+	exit -12
+fi
+
+
+#taken from professor Thain's makeflow_ec2_setup
+if which aws >/dev/null 2>&1
+then
+	echo "found aws, moving on to config file check"
+else
+	echo "aws not found in path, please include it"
+	exit -9
+fi
+
+if [ -f ~/.aws/config ]; then
+	echo "found config file, moving on to credentials file check"
+else
+	echo "aws not configured: please configure before running the script"
+	exit -10
+fi
+if [ -f ~/.aws/credentials ]; then
+	echo "found credentials file!"
+else
+	echo "aws not configured: please configure before running the script"
+	exit -10
+fi
+
+if aws batch list-jobs --job-queue HighPriority > /dev/null 2>&1
+then
+	echo "config and credentials configured correctly, proceeding to setup"
+else
+	echo "configuration and credentials files not properly set, please reconfig aws and try again"
+	exit -33
+fi
+
+check_arn_iam=$(aws --output json iam get-role --role-name AWSBatchServiceRole)
+#good_iam=$(python -c "print 'True' if 'CreateDate' in '$check_arn_iam' else print 'False'")
+if [[ $check_arn_iam = *"CreateDate"*  ]]; then
```

Comment:
Double [ is bash. Use a single [ for sh. 

That matching operator may not be available in sh, though, so you may need to find another solution.